### PR TITLE
Added new xml libs

### PR DESCRIPTION
**Pull request by jaredsburrows**
@mikepenz 
- Added new libs:
```
	library/src/main/res/values/library_appcompat_v7_strings.xml
	library/src/main/res/values/library_design_strings.xml
	library/src/main/res/values/library_okio_strings.xml
	library/src/main/res/values/library_recyclerview_v7_strings.xml
	library/src/main/res/values/library_support_annotations_strings.xml
	library/src/main/res/values/library_support_v4_strings.xml
```
- Updated to Gradle `2.12`
- Fixed file permissions
 - folders - 755
 - files - 644
- Removed `/libs` folder in favor of AAR

### CODE REVIEW

File: library/src/main/res/values/library_retrofit_strings.xml [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
     <string name="library_Retrofit_libraryDescription">A type-safe REST client for Android and Java.
     </string>
-    <string name="library_Retrofit_libraryVersion">1.9.0</string>
+    <string name="library_Retrofit_libraryVersion">2.1</string>
     <string name="library_Retrofit_libraryWebsite">http://square.github.io/retrofit/</string>
     <string name="library_Retrofit_licenseId">apache_2_0</string>
     <string name="library_Retrofit_isOpenSource">true</string>
     <string name="library_Retrofit_repositoryLink">https://github.com/square/retrofit</string>
-    <string name="library_Retrofit_classPath">retrofit.Utils</string>
+    <string name="library_Retrofit_classPath">retrofit2</string>
```
Comment (mikepenz) [inline library/src/main/res/values/library_retrofit_strings.xml]: This requires a class which the `detector` tries to init. `retrofit2.Utils` should be fine

File: library/src/main/res/values/library_support_annotations_strings.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	<!-- Library section -->
+	<string name="library_support_annotations_libraryName">support-annotations</string>
+	<string name="library_support_annotations_libraryDescription">The Annotation package provides APIs to support adding annotation metadata to your apps.</string>
+	<string name="library_support_annotations_libraryWebsite"></string>
+	<string name="library_support_annotations_libraryVersion">23.2.1</string>
+	<!-- OpenSource section -->
+	<string name="library_support_annotations_isOpenSource">true</string>
+	<string name="library_support_annotations_repositoryLink"></string>
+	<!-- ClassPath for autoDetect section -->
+	<string name="library_support_annotations_classPath">android.support.annotation</string>
```
Comment (mikepenz) [inline library/src/main/res/values/library_support_annotations_strings.xml]: not required in this case.

File: library/src/main/res/values/library_support_v4_strings.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	<!-- Library section -->
+	<string name="library_support_v4_libraryName">support-v4</string>
+	<string name="library_support_v4_libraryDescription">This library is designed to be used with Android 1.6 (API level 4) and higher. It includes the largest set of APIs compared to the other libraries, including support for application components, user interface features, accessibility, data handling, network connectivity, and programming utilities.</string>
+	<string name="library_support_v4_libraryWebsite">https://developer.android.com/tools/support-library/features.html#v4</string>
+	<string name="library_support_v4_libraryVersion">23.2.1</string>
+	<!-- OpenSource section -->
+	<string name="library_support_v4_isOpenSource">true</string>
+	<string name="library_support_v4_repositoryLink"></string>
+	<!-- ClassPath for autoDetect section -->
+	<string name="library_support_v4_classPath">android.support.v4</string>
```
Comment (mikepenz) [inline library/src/main/res/values/library_support_v4_strings.xml]: also not necessary

File: library/src/main/res/values/library_recyclerview_v7_strings.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	<!-- Library section -->
+	<string name="library_recyclerview_v7_libraryName">recyclerview-v7</string>
+	<string name="library_recyclerview_v7_libraryDescription">The recyclerview library adds the RecyclerView class. This class provides support for the RecyclerView widget, a view for efficiently displaying large data sets by providing a limited window of data items.</string>
+	<string name="library_recyclerview_v7_libraryWebsite"></string>
+	<string name="library_recyclerview_v7_libraryVersion">23.2.1</string>
+	<!-- OpenSource section -->
+	<string name="library_recyclerview_v7_isOpenSource">true</string>
+	<string name="library_recyclerview_v7_repositoryLink"></string>
+	<!-- ClassPath for autoDetect section -->
+	<string name="library_recyclerview_v7_classPath">android.support.v7.recyclerview</string>
```
Comment (mikepenz) [inline library/src/main/res/values/library_recyclerview_v7_strings.xml]: also not necessary

File: library/src/main/res/values/library_design_strings.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	<!-- Library section -->
+	<string name="library_design_libraryName">design</string>
+	<string name="library_design_libraryDescription">The Design package provides APIs to support adding material design components and patterns to your apps.  The Design Support library adds support for various material design components and patterns for app developers to build upon, such as navigation drawers, floating action buttons (FAB), snackbars, and tabs.</string>
+	<string name="library_design_libraryWebsite"></string>
+	<string name="library_design_libraryVersion">23.2.1</string>
+	<!-- OpenSource section -->
+	<string name="library_design_isOpenSource">true</string>
+	<string name="library_design_repositoryLink"></string>
+	<!-- ClassPath for autoDetect section -->
+	<string name="library_design_classPath">android.support.design</string>
```
Comment (mikepenz) [inline library/src/main/res/values/library_design_strings.xml]: also not necessary

File: library/src/main/res/values/library_appcompat_v7_strings.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	<!-- Library section -->
+	<string name="library_appcompat_v7_libraryName">appcompat-v7</string>
+	<string name="library_appcompat_v7_libraryDescription">This library adds support for the Action Bar user interface design pattern.</string>
+	<string name="library_appcompat_v7_libraryWebsite"></string>
+	<string name="library_appcompat_v7_libraryVersion">23.2.1</string>
+	<!-- OpenSource section -->
+	<string name="library_appcompat_v7_isOpenSource">true</string>
+	<string name="library_appcompat_v7_repositoryLink"></string>
+	<!-- ClassPath for autoDetect section -->
+	<string name="library_appcompat_v7_classPath">android.support.v7.appcompat</string>
```
Comment (mikepenz) [inline library/src/main/res/values/library_appcompat_v7_strings.xml]: also not necdssary

**Comment by mikepenz** [comment]
@jaredsburrows thank you so much. Next time I would prefer split up pull request, per change. So it is easier to check what changed and what to merge. 

I also checked all changes. For `retrofit2` the `classPath` needs a change, and for the support libraries I would prefer to not have the `classPath` defined, as it would otherwise automatically show the lib within the about activity, and I am not sure if this is the prefered solution for most. What do you think?

**Comment by jaredsburrows** [comment]
@mikepenz Next time I will split up the PR.

I have updated retrofit2.

The support libraries are not automatically shown in the about activity. This is why I am adding them.

**Comment by mikepenz** [comment]
@jaredsburrows oh this is now in conflict with the merge of the ci pull request.

**Comment by jaredsburrows** [comment]
@mikepenz Fixed. Let me know if all is well.

**Comment by mikepenz** [comment]
@jaredsburrows fine so far. only issue now is that the `buildPaths` should be a specific class which the `AboutLibrary` can try to instantiate. Will change it on my side. Thank you.

**Comment by jaredsburrows** [comment]
@mikepenz `buildPaths` or `classPath`? Sorry about that. Now I understand what you are asking. In the future I will make sure I have a `class` specified and not a `package.

**Comment by mikepenz** [comment]
@jaredsburrows oops I meant the `classPath`. the `classPath` name may be a bit missleading, as you correctly defined a `classPath`, but as far as my research has found, there is no possibility to just check via a packagename if it exists. The only way is to try to instantiate a `class`

**Pull request merged**